### PR TITLE
Generate "install dependencies" docs from CI script

### PR DESCRIPTION
**Pull request by jpreiss**
We maintain lists of Ubuntu (and now Pip) packages for dependencies in two different places: the CI script and the docs. This creates an opportunity for them to diverge (e.g. 2c3e789a637d8c41e1a1906faaae1dd7dce1a6e4 could have also removed `sdcc` from the CI version).

In #342 we considered and rejected the idea of somehow running the CI script on the local machine. It is too complicated.

In #342 we also discussed the idea of moving the dependency install into a shell script. This is reasonable, but I prefer showing the list of installed packages in the docs for transparency.

Therefore, we instead generate the docs from the CI script in this PR.

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -14,6 +14,15 @@ ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) .
 # the i18n builder cannot share the environment and doctrees with the others
 I18NSPHINXOPTS  = $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) .
 
+# Generated docs.
+generated/full_dependencies.rstinclude: ../.github/workflows/ci-ros.yml generate_install_deps_code.py
+	python $(word 2,$^) $< > $@
```
Comment (whoenig) [inline docs/Makefile:19]: Is that a problem with Ubuntu 20.04 (only python3 exists by default)?

**Review by whoenig — APPROVED** [review]
Very neat idea!

**Comment by jpreiss** [comment]
Ah, this fails on readthedocs because it doesn't actually use the Makefile - it appears to run Sphinx directly.

Is there a way to configure readthedocs to run on PRs or all branches?

**Comment by whoenig** [comment]
That's a bummer:-( I did find a setting to enable building on PRs and switched that on.

**Comment by jpreiss** [comment]
It's OK, I found a sphinx extension for including program output in the docs.

**Pull request merged**